spring-integration-ip/src/test/java/org/springframework/integration/ip/tcp/connection/SocketSupportTests.java

```diff
@@ -318,11 +331,20 @@ public void testNetClientAndServerSSLDifferentContexts() throws Exception {
 			latch.countDown();
 			return false;
 		});
+		server.setTcpSocketSupport(new DefaultTcpSocketSupport() {
+
+			@Override
+			public void postProcessServerSocket(ServerSocket serverSocket) {
+				((SSLServerSocket) serverSocket).setNeedClientAuth(true);
```

Comment:
By default, only the server certificate is checked. Client auth is either not required, requested (optional) or required. See [the javadocs](https://docs.oracle.com/javase/8/docs/api/javax/net/ssl/SSLSocket.html#setNeedClientAuth-boolean-).